File: training-data/dart/pair_sum_sorted.dart
(int, int)? pairWithSum(List<int> nums, int target) {
  var i = 0, j = nums.length - 1;
  while (i < j) {
    final s = nums[i] + nums[j];
    if (s == target) return (i, j);
    if (s < target) {
      i++;
    } else {
      j--;
    }
  }
  return null;
}

void main() {
  print(pairWithSum([1, 3, 4, 6, 8, 11], 10));
  print(pairWithSum([1, 2, 3], 100));
}

File: training-data/lua/pair_sum_sorted.lua
local function pair_with_sum(nums, target)
  local i, j = 1, #nums
  while i < j do
    local s = nums[i] + nums[j]
    if s == target then return i, j end
    if s < target then i = i + 1 else j = j - 1 end
  end
  return nil
end

print(pair_with_sum({ 1, 3, 4, 6, 8, 11 }, 10))
print(pair_with_sum({ 1, 2, 3 }, 100))
